File: 2022/02/0202.py
f = open("input.txt", "r")

# A = 1 (rock)
# B = 2 (paper)
# C = 3 (scissors)

# W, Z = 6
# D, Y = 3
# L, X = 0

x = { 'B': 1, 'A': 3, 'C': 2 } # loss
y = { 'C': 3, 'B': 2, 'A': 1 } # draw
z = { 'A': 2, 'C': 1, 'B': 3 } # win

def main():
    score = 0;
    for x in f:
        match = x.replace("\n", "").split(" ");
        score += getScore(match)
    print(score);
        

def getScore(match:list[int]) -> int:
    match match[1]:
        case 'X':
            return x[match[0]] + 0;
        case 'Y':
            return y[match[0]] + 3;
        case 'Z':
            return z[match[0]] + 6;
        case _:
            print("No matches wuuuut???");

main();

File: 2022/02/0202.kt
import java.io.File

val x = mapOf("B" to 1, "A" to 3, "C" to 2)
val y = mapOf("C" to 3, "B" to 2, "A" to 1)
val z = mapOf("A" to 2, "C" to 1, "B" to 3)
fun main() {

    var score: Int = 0

    File("input.txt").forEachLine { it ->
        var match = it.replace("\n", "").split(" ").toTypedArray()
        score += getScore(match)
    }
    println(score)
}

fun getScore(match: Array<String>): Int {
    when (match[1]) {
        "X" -> {
            var score = x[match[0]]
            if (score != null) {
                return score + 0
            }
        }
        "Y" -> {
            var score = y[match[0]]
            if (score != null) {
                return score + 3
            }
        }
        "Z" -> {
            var score = z[match[0]]
            if (score != null) {
                return score + 6
            }
        }
        else -> {
            println("wut")
        }
    }
    return -1
}